### PR TITLE
Update Python libraries

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
A number of Python libraries have slowly grown outdated. Other libraries were listed twice. While still, some were no longer used (if they were ever used in the first place)

Notable updates include:

- `black`
- `boto3`
- `certifi`
- `docker`
- `flake8`
- `pandas`
- `pip`
- `psycopg2-binary`
- `pytest`
- `requests`
- `xlrd` -> `xlrd3`

After the infrastructure migration is complete, upgrading from Django 2.2 to 3.1 should be considered since it should be straightforward and will make several desired features available.

**Technical details:**
Many packages from the requirements* files were removed since they were dependencies for other libraries. By removing them from the files, pip can upgrade them while maintaining compatibility with the rest of the installs. While this should both reduce developer overhead and help keep the libraries versions automatically up-to-date, there are small risks of needing to manually intervene if a package version is flagged and isn't being automatically upgraded due to compatibility issues.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated (N/A)
2. [x] API documentation updated (N/A)
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Operations
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed (N/A)
7. [ ] Appropriate Operations ticket(s) created
8. [x] Jira Ticket (N/A)

**Area for explaining above N/A when needed:**
```
No changes to API, some code was modified to meet latest static check expectations
```

### CODE REVIEW

**Review by brett-varney — COMMENTED** [review]
Quick read-through: I don't see anything but trivial changes, but there's a lot. A few code updates, like bracketing values with a list(), and I'm not sure what would be a full-confidence test of all these package updates.

**Comment by tony-sappe** [comment]
> Quick read-through: I don't see anything but trivial changes, but there's a lot. A few code updates, like bracketing values with a list(), and I'm not sure what would be a full-confidence test of all these package updates.

- Bracketing values with a list() was necessary for the change from `xlrd` -> `xlrd3`
- A few single-character variable names were manually changed to satisfy new `flake8` rules
- All docstring changes were by `black`
- A number of multiline statements were altered by black to be spread out across even more lines. The culprit was an unnecessary trailing comma. Manually removing the trailing commas was the minimal code change to satisfy the latest version of black.

A majority of the library version changes did not require any code changes. Next steps of testing involves a deployed environment (dev or sandbox)

**Comment by tony-sappe** [comment]
We're going to wait for the additional testing burden of the infrastructure migration is past before merging this PR

File: requirements/requirements-app.txt [hunk 1]
```diff
@@ -1,62 +1,38 @@
...
-boto3==1.9.224
-botocore==1.12.224
-certifi==2019.6.16
-chardet==3.0.4
+attrs==20.*
+boto3==1.16.*
+certifi==2020.12.5
 dataclasses-json==0.5.*
-django_cte==1.1.*
 ddtrace==0.37.1
```
Comment (brett-varney) [inline requirements/requirements-app.txt:7]: This version doesn't work on (my?) Win10; is there a reason not to bump it to 0.40?

**Review by brett-varney — COMMENTED** [review]
ddtrace version

File: requirements/requirements-app.txt (hunk 1 above)
Comment (tony-sappe) [reply requirements/requirements-app.txt:7]: @kwhickey @bminahan-kc @gpontejos Any input on a newer version of DDtrace?

Comment (bminahan-kc) [reply requirements/requirements-app.txt:7]: .40 should be fine. The latest is 0.45 if we wanna go there. We use dd-agent version 6.15 currently which looks compatible.

Comment (tony-sappe) [reply requirements/requirements-app.txt]: Changed to 0.45.x

**Review by sethstoudenmier — COMMENTED** [review]
Tested locally and appears to be working as expected with no change in functionality.

I do have one question before approving. Did you run into needing to run a migration for the following:
>  [ ] 0006_view_and_view_method_nullable
 [ ] 0006_auto_20180315_1442
 [ ] 0007_merge_20180419_1646
 [ ] 0008_auto_20200201_2048
 [ ] 0009_view_method_max_length_200
 [ ] 0010_auto_20200609_1404
 [ ] 0011_auto_20201117_2016

After checking out the branch I was prompted to run a migration for these listed above. Looks like it is due to a change in `drf-tracking`.

Will need to add a step to the deploy coordination if these migrations are a result of these changes.

**Comment by tony-sappe** [comment]
> Tested locally and appears to be working as expected with no change in functionality.
> 
> I do have one question before approving. Did you run into needing to run a migration for the following:
> 
> > [ ] 0006_view_and_view_method_nullable
> > [ ] 0006_auto_20180315_1442
> > [ ] 0007_merge_20180419_1646
> > [ ] 0008_auto_20200201_2048
> > [ ] 0009_view_method_max_length_200
> > [ ] 0010_auto_20200609_1404
> > [ ] 0011_auto_20201117_2016
> 
> After checking out the branch I was prompted to run a migration for these listed above. Looks like it is due to a change in `drf-tracking`.
> 
> Will need to add a step to the deploy coordination if these migrations are a result of these changes.

https://federal-spending-transparency.atlassian.net/browse/DEV-6729

**Review by sethstoudenmier — APPROVED** [review]

**Pull request merged**